diff -rupb linux-5.9.1-orig/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64-plus.dts linux-5.9.1/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64-plus.dts
--- linux-5.9.1-orig/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64-plus.dts	2020-11-02 21:33:56.533174632 +0100
+++ linux-5.9.1/arch/arm64/boot/dts/allwinner/sun50i-a64-pine64-plus.dts	2020-11-02 21:28:06.050766400 +0100
@@ -16,6 +16,8 @@
 	phy-mode = "rgmii";
 	phy-handle = <&ext_rgmii_phy>;
 	status = "okay";
+        allwinner,tx-delay-ps = <500>;
+        allwinner,rx-delay-ps = <500>;
 };
 
 &mdio {
