--- linux-5.9.1-orig/arch/arm64/boot/dts/allwinner/sun50i-a64-sopine-baseboard.dts	2020-10-17 08:31:22.000000000 +0200
+++ linux-5.9.1/arch/arm64/boot/dts/allwinner/sun50i-a64-sopine-baseboard.dts	2020-11-03 18:50:15.932922190 +0100
@@ -79,7 +79,7 @@
 &emac {
 	pinctrl-names = "default";
 	pinctrl-0 = <&rgmii_pins>;
-	phy-mode = "rgmii";
+	phy-mode = "rgmii-txid";
 	phy-handle = <&ext_rgmii_phy>;
 	phy-supply = <&reg_dc1sw>;
 	status = "okay";
