--- boinc-client_release-7.16-7.16.11/client/Makefile.am	2020-08-30 10:04:59.000000000 +0200
+++ 7.16.11/client/Makefile.am	2020-11-03 17:52:25.971909691 +0100
@@ -115,20 +115,6 @@ boinc_client_LDADD = $(LIBBOINC) $(LIBBO
 
 boinc_clientdir = $(bindir)
 
-if OS_ARM_LINUX
-boinccmd_CPPFLAGS += -mfpu=vfpv3-d16
-boinc_client_CPPFLAGS += -mfpu=vfpv3-d16
-boinc_client_CXXFLAGS += -mfpu=vfpv3-d16
-EXTRA_boinc_client_DEPENDENCIES = libwhetneon.a libwhetvfp.a
-boinc_client_LDADD += libwhetneon.a libwhetvfp.a
-noinst_LIBRARIES = libwhetneon.a libwhetvfp.a
-libwhetneon_a_SOURCES = whetstone.cpp
-libwhetneon_a_CXXFLAGS = $(filter-out -mfpu=vfpv3-d16,$(boinc_client_CXXFLAGS)) -DANDROID_NEON -mfloat-abi=softfp -mfpu=neon
-
-libwhetvfp_a_SOURCES = whetstone.cpp
-libwhetvfp_a_CXXFLAGS = $(boinc_client_CXXFLAGS) -DANDROID_VFP -mfloat-abi=softfp -mfpu=vfpv3-d16
-endif
-
 switcher_SOURCES = switcher.cpp
 switcher_LDFLAGS = $(AM_LDFLAGS) -L../lib
 switcher_LDADD = $(LIBBOINC)
