--- a/client/hostinfo_unix.cpp	2020-12-02 15:04:22.000000000 +0100
+++ b/client/hostinfo_unix.cpp	2021-01-06 21:48:23.691494483 +0100
@@ -41,6 +41,10 @@
 #include <cstring>
 #endif
 
+#ifdef __GLIBC__
+#include <gnu/libc-version.h>
+#endif
+
 #if HAVE_XSS
 #include <X11/extensions/scrnsaver.h> //X-based idle detection
 // prevents naming collision between X.h define of Always and boinc's
@@ -1371,6 +1375,12 @@ int HOST_INFO::get_memory_info() {
 // return BOINC_SUCCESS if at least version could be found (extra_info may remain empty)
 // return ERR_NOT_FOUND if ldd couldn't be opened or no version information was found
 //
+#ifdef __GLIBC__
+int get_libc_version(string& version, string& extra_info) {
+    version = string(gnu_get_libc_version());
+    return BOINC_SUCCESS;
+}
+#else
 int get_libc_version(string& version, string& extra_info) {
     char buf[1024] = "";
     string strbuf;
@@ -1406,6 +1416,7 @@ int get_libc_version(string& version, st
     return BOINC_SUCCESS;
 }
 #endif
+#endif
 
 // get os_name, os_version
 //
