diff --git a/client/cpu_sched.cpp b/client/cpu_sched.cpp
index bce568d608..37ada7c488 100644
--- a/client/cpu_sched.cpp
+++ b/client/cpu_sched.cpp
@@ -79,6 +79,8 @@
 #include "project.h"
 #include "result.h"
 
+#define BACKOFF 140 // 2 minutes and 20 seconds
+
 
 using std::vector;
 using std::list;
@@ -146,7 +148,13 @@ struct PROC_RESOURCES {
                 atp->needs_shmem = false;
             }
         }
-        if (rp->schedule_backoff > gstate.now) return false;
+        if (rp->schedule_backoff > gstate.now) {
+            if(log_flags.cpu_sched_debug) {
+                msg_printf(rp->project, MSG_INFO, 
+                "[cpu_sched_debug] there's a backoff on this project... not scheduling");
+            }
+            return false;
+        }
         if (rp->uses_gpu()) {
             if (gpu_suspend_reason) return false;
         }
@@ -1308,11 +1316,13 @@ bool CLIENT_STATE::enforce_run_list(vector<RESULT*>& run_list) {
             if (atp) {
                 atp->too_large = true;
             }
+            rp->schedule_backoff = gstate.now + BACKOFF;
             if (log_flags.cpu_sched_debug || log_flags.mem_usage_debug) {
                 msg_printf(rp->project, MSG_INFO,
-                    "[cpu_sched_debug] enforce: task %s can't run, too big %.2fMB > %.2fMB",
-                    rp->name,  wss/MEGA, ram_left/MEGA
+                    "[cpu_sched_debug] enforce: task %s can't run, too big %.2fMB > %.2fMB, setting backoff to %.2f",
+                    rp->name,  wss/MEGA, ram_left/MEGA, rp->schedule_backoff
                 );
+                
             }
             continue;
         }
diff --git a/client/gui_rpc_server_ops.cpp b/client/gui_rpc_server_ops.cpp
index 921b20abbf..ef35f12a7c 100644
--- a/client/gui_rpc_server_ops.cpp
+++ b/client/gui_rpc_server_ops.cpp
@@ -1960,6 +1960,7 @@ int GUI_RPC_CONN::handle_rpc() {
             "Server: BOINC client\n"
             "Access-Control-Allow-Origin: *\n"
             "Access-Control-Allow-Methods: POST, GET, OPTIONS\n"
+            "Access-Control-Allow-Headers: *\n"
             "Content-Length: 0\n"
             "Keep-Alive: timeout=2, max=100\n"
             "Connection: Keep-Alive\n"
@@ -2046,6 +2047,9 @@ int GUI_RPC_CONN::handle_rpc() {
             "HTTP/1.1 200 OK\n"
             "Date: Fri, 31 Dec 1999 23:59:59 GMT\n"
             "Server: BOINC client\n"
+            "Access-Control-Allow-Origin: *\n"
+            "Access-Control-Allow-Methods: POST, GET, OPTIONS\n"
+            "Access-Control-Allow-Headers: *\n"
             "Connection: close\n"
             "Content-Type: text/xml; charset=utf-8\n"
             "Content-Length: %d\n\n"
